cmd/containerd-shim-runhcs-v1/pod.go

```diff
@@ -141,29 +148,68 @@ func createPod(ctx context.Context, events publisher, req *task.CreateTaskReques
 		id:     req.ID,
 		host:   parent,
 	}
-	// TOOD: JTERRY75 - There is a bug in the compartment activation for Windows
+
+	caAddr := fmt.Sprintf(addrFmt, parent.ID())
+	if parent != nil && parent.NCProxyEnabled() {
+		// Setup compute agent service
+		ttrpcListener, err := winio.ListenPipe(caAddr, nil)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to listen on %s", caAddr)
+		}
+		s, err := ttrpc.NewServer(ttrpc.WithUnaryServerInterceptor(octtrpc.ServerInterceptor()))
+		if err != nil {
+			return nil, err
+		}
+		caService := &computeAgent{parent}
+		computeagent.RegisterComputeAgentService(s, caService)
+		serveComputeAgent(ctx, s, ttrpcListener)
+	}
+
+	// TODO: JTERRY75 - There is a bug in the compartment activation for Windows
 	// Process isolated that requires us to create the real pause container to
 	// hold the network compartment open. This is not required for Windows
 	// Hypervisor isolated. When we have a build that supports this for Windows
 	// Process isolated make sure to move back to this model.
+
+	// For WCOW we fake out the init task since we dont need it. We only
+	// need to provision the guest network namespace if this is hypervisor
+	// isolated. Process isolated WCOW gets the namespace endpoints
+	// automatically.
+	nsid := ""
 	if isWCOW && parent != nil {
-		// For WCOW we fake out the init task since we dont need it. We only
-		// need to provision the guest network namespace if this is hypervisor
-		// isolated. Process isolated WCOW gets the namespace endpoints
-		// automatically.
-		if parent != nil {
-			nsid := ""
-			if s.Windows != nil && s.Windows.Network != nil {
-				nsid = s.Windows.Network.NetworkNamespace
-			}
+		if s.Windows != nil && s.Windows.Network != nil {
+			nsid = s.Windows.Network.NetworkNamespace
+		}
 
-			if nsid != "" {
+		if nsid != "" {
+			// If client passed in ncproxy addr, check if client is not nil and setup
+			// the network using the proxy.
+			if parent.NCProxyEnabled() {
```

Comment:
Rather than needing to conditionally switch our behavior here, we should instead try to have multiple implementations of a common "network configuration" interface, and choose the correct implementation (standard or ncproxy) when the shim starts up. If we can re-use the `NodeNetworkServiceServer` interface for this, that would be good.

Reply:
Oh I like that, I'll play around with it